main.py

```diff
@@ -40,3 +47,107 @@ def shutdown(self):
+            if mw is None:
+                result = 'One or more items are invalid'
+                status = 400
+            elif mw.start < now:
+                result = 'Start in the past not allowed'
+                status = 400
+            elif mw.end <= mw.start:
+                result = 'End before start not allowed'
+                status = 400
+            else:
+                self.scheduler.add(mw)
+                self.maintenances[mw.id] = mw
+                result = {'mw_id': mw.id}
+                status = 201
+        return jsonify(result), status
+
+    @rest('/<mw_id>', methods=['PATCH'])
+    def update_mw(self, mw_id):
+        """Update a maintenance window."""
+        data = request.get_json()
+        if not data:
+            result = "Bad request: The request do not have a json."
+            status = 415
+            log.debug('update_mw result %s %s', result, status)
+        else:
+            try:
+                mw = self.maintenances[mw_id]
+                mw.update(data, self.controller)
+                result = {'response': f'Maintenance {mw_id} updated'}
+                status = 201
+            except KeyError:
+                result = {'response': f'Maintenance with id {mw_id} not found'}
+                status = 404
+        return jsonify(result), status
+
+    @rest('/<mw_id>', methods=['DELETE'])
+    def remove_mw(self, mw_id):
+        """Delete a maintenance window."""
+        try:
+            mw = self.maintenances[mw_id]
+            self.scheduler.remove(mw)
+            del self.maintenances[mw_id]
+            result = {'response': f'Maintenance with id {mw_id} successfully '
+                                  f'removed'}
+            status = 200
+        except KeyError:
+            result = {'response': f'Maintenance with id {mw_id} not found'}
+            status = 404
+        return jsonify(result), status
+
+    @rest('/<mw_id>/end', methods=['PATCH'])
+    def end_mw(self, mw_id):
+        """Finish a maintenance window right now."""
+        try:
+            mw = self.maintenances[mw_id]
+        except KeyError:
+            result = {'response': f'Maintenance with id {mw_id} not found'}
+            status = 404
+        else:
+            now = datetime.datetime.now(pytz.utc)
+            log.info(f'Agora {now}')
+            if now < mw.start:
+                result = {'response': f'Maintenance window {mw_id} has not '
+                                      f'yet started.'}
+                status = 400
+            elif now > mw.end:
+                result = {'response': f'Maintenance window {mw_id} has '
+                                      f'already finished.'}
+                status = 400
+            else:
+                self.scheduler.remove(mw)
+                mw.end_mw()
+                result = {'response': f'Maintenance window {mw_id} finished.'}
+                status = 200
+        return jsonify(result), status
```

Comment:
For all the REST methods I'd suggest copying the `return jsonify(...)` to the top of the function (inside the error code path) and moving all the "default" code path to outside the `else` block.

> flat is better than nested